### PR TITLE
DP-15349 MF Add text fade to service page overview for flexible link groups

### PR DESCRIPTION
**Pull request by lannyheidbreder-current**
## Description
Added collapsible/expandable text fade to overview paragraphs on ServiceFlexibleLinkGroups page component.

## Related Issue / Ticket

- [JIRA issue](https://jira.mass.gov/browse/DP-15349)

#### Go here and do this

In Mayflower, go to `?p=pages-service-flexible-link-groups` in a sufficiently narrow browser window. (970px or smaller should suffice.)

#### What you should see

The intro text is truncated and fades out toward the bottom, and a button matching the design spec allows you to expand and collapse the area.

At browser widths above 840px (yet small enough to trigger the truncation at all), six lines of text (or equivalent height) should be fully or partially visible.

At 840px or below, four lines of text (or equivalent height) should be fully or partially visible.

## Screenshots

Screen Shot 2019-09-10 at 8 22 07 AM
Screen Shot 2019-09-10 at 8 22 26 AM

## Additional Notes:

I think we will need to add some classes to `activateBodyTextTruncation.js` to affect mass.gov properly; the classes I'm targeting are Mayflower-only, I suspect. I don't think it will require actual back-end work, though I could be wrong about that.

When this hits develop, [DP-8847](https://jira.mass.gov/browse/DP-8847) "should" be quick and easy to finish.

See my assessment of the design spec in the comment on the JIRA ticket.

#### Impacted Areas in Application

A single Mayflower component, for now.

### CODE REVIEW

**Review by ygannett — CHANGES_REQUESTED** [review]
This is working great!

I have a couple of things to mention for accessibility and design.

**Accessibility**:  The "Show more/less" button doesn't give any context to screen reader users. They wouldn't know WHAT to show more/less as they reach the button.  Let's add **aria-label** to `<button>` to tell what to show/hide.  Using the words "show" and "hide" for screen reader users since they don't see the visual presentation and they reach the button before the associated content.  Its value should reflects its state.
       - Visible label = "Show more" --> aria-label="Show the overview."
       - Visible label = "Show less" --> aria-label="Hide the overview."

**Design** :  The alignment of the button doesn't follow our existing design pattern. The design team pointed to this pattern:  https://mayflower.digital.mass.gov/?p=organisms-sticky-toc
Pattern_Lab_-_organisms-sticky-toc
Let's set left align the button.

**Comment by lannyheidbreder-current** [comment]
@ygannett, the button alignment is as requested in this ticket’s original mockup; any changes would be considered a scope change and would require review by a designer or manager and so would require a new ticket. Also, per Joe's request that accessibility updates to patterns be tracked appropriately, please create a new ticket for those changes.

**Comment by joeg8612** [comment]
@lannyheidbreder-mc I think that the accessibility change requested is directly related to the code written so is in scope for this PR.   I am fine with any design changes being ticketed as a potential change for later.  The design team spec'ed out this design and reviewed the implementation on my laptop.   cc @ygannett

**Review by ygannett — APPROVED** [review]
It looks great.  Thank you for all work.

File: assets/scss/00-base/mixins/_ma-truncated-body-text.scss [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  .ma-truncated-body-text__fade-overlay {
+    position: absolute;
+    z-index: 2;
+    width: 100%;
+    height: 60px;
+    left: 0;
+    bottom: 0;
+    background: linear-gradient(
+      to bottom,
+      rgba(#fff, 0) 0%,
```
Comment (avrilpearl) [inline assets/scss/00-base/mixins/_ma-truncated-body-text.scss]:
```suggestion
      rgba($c-white, 0) 0%,
```

File: assets/scss/00-base/mixins/_ma-truncated-body-text.scss [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      rgba(#fff, 1) 100%
```
Comment (avrilpearl) [inline assets/scss/00-base/mixins/_ma-truncated-body-text.scss]:
```suggestion
      rgba($c-white, 1) 100%
```

File: assets/scss/00-base/mixins/_ma-truncated-body-text.scss [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    @include ma-chevron;
+    order: 2;
+    width: 100%;
+    position: relative;
+    z-index: 3;
+    height: 60px;
+    background: transparent;
+    border: 0;
+    font-weight: 700;
+    color: $c-bay-blue;
```
Comment (avrilpearl) [inline assets/scss/00-base/mixins/_ma-truncated-body-text.scss]:
```suggestion
    color: $c-primary;
```

File: patternlab/styleguide/source/assets/js/helpers/listing.js [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import getTemplate from "../helpers/getHandlebarTemplate.js";
 import sticky from "../helpers/sticky.js";
 import getOuterHtml from "../helpers/getElementOuterHtml.js";
+import activateBodyTextTruncation from "../modules/activateBodyTextTruncation.js";
```
Comment (avrilpearl) [inline patternlab/styleguide/source/assets/js/helpers/listing.js]: not sure why this import is here, seems to be removable:
```suggestion
```

File: assets/scss/00-base/mixins/_ma-truncated-body-text.scss [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+    &::after {
+      transform: translateY(-35%) rotate(45deg);
+      opacity: 0.5;
```
Comment (avrilpearl) [inline assets/scss/00-base/mixins/_ma-truncated-body-text.scss]: i think this should not have an opacity but be $c-primary-lightest;

**Review by avrilpearl — COMMENTED** [review]
still working on review js locally

**Review by avrilpearl — APPROVED** [review]

**Pull request merged**